### ISSUE TITLE
Cache lists used for widgets

### ISSUE THREAD

**Issue by juanma-cvega** [jurialmunkey/plugin.video.themoviedb.helper#647]
I currently use many different lists for my widgets, like my Trakt lists, Trakt liked ones, in progress, trending or popular or TMDB discover lists. I'm using an Nvidia Shield 2019 Pro and I've noticed with the recent changes that things have sped up nicely but it always takes several minutes to load Kodi just because of these widgets. I use 4 different groups (movies, TV shows, anime and kids), each one with several widgets each. 
Would there be a way to mark lists used for widgets so all the information is cached (images and information) so they are loaded faster? I was thinking maybe these lists could be refreshed automatically in the background from time to time given they don't actually change that often. I'm using Arctic Horizon as skin and it tends to refresh widgets quite often, either because Kodi crashed and I have to restart it or after watching something. I haven't identified yet what makes the groups reload the widgets but every time this happens it takes a while just to load them.

**Comment by jurialmunkey** [comment]
The data for TMDb lists is already cached (typically for one day) and the detailed information for each item is kept for 14 days. Trakt lists like in-progress and up next are cached and only refresh when the trakt last activities timestamp changes. Other Trakt lists do refresh every time the first page is opened due to needing to resort, so it really depends on the list.

Do you have any specific lists in mind? Also helpful would be if you turn on Timer Reports in TMDbH Expert settings and then get me the log of you opening a few of the lists inside TMDbH one at a time (i.e. not as widgets) as that will help me see exactly what areas are problematic if it is actually the retrieval of the list data or it is something else.

Also, I've noticed slow Trakt response times recently which definitely isn't helping.

>  I haven't identified yet what makes the groups reload the widgets but every time this happens it takes a while just to load them.

Kodi reloads all containers whenever you stop playback, after a library update completes, when you reload the skin. There are a couple of other scenarios but they typically don't apply to widgets.

Skins typically don't control when widgets reload. There are some tricks they can use to force widgets to update but Arctic Horizon doesn't do anything like that. Any widget reloading that happens in AH is triggered by Kodi or an add-on.

**Comment by jurialmunkey** [comment]
^^^ ~v4.5.73 adds an option "Always refresh Trakt custom lists" under `TMDbH Settings > Accounts` which will be enabled by default but you can disable to turn off the first page refresh on custom lists. With that disabled the list should be cached for 1 day and should help with reloading widgets if you use a lot of Trakt custom lists either made by yourself or other people (e.g. the ones in Popular Lists or Trending Lists)~

EDIT: Changed my mind. I think this setting is unnecessary and should be handled automatically. All Trakt custom lists will now automatically be cached for one day _except_ lists that you made yourself. Lists that you made yourself will still refresh every time so that it will update properly when you add/remove/reorder items.

This should improve reloading times a fair bit if you use a lot of Trakt custom lists made by other people (e.g. all the lists in the "Popular Lists" section etc.)

**Comment by matke-84** [comment]
@jurialmunkey
I did one test. Compared how fast the tmdb helper widget loads and how fast the elementum widget. I was very surprised by the testing. The tmdb helper widget loads a few seconds faster than the elementum. I am surprised because the same lists in the addons themselves elementum open faster. I'm not talking about cached data here. I tested like that I deleted all cache from tmdb helper and elementum, as well from kodi, thumbs, turned on all artworks from fanarts.tv... So it's like opening kodi and addons for the first time. After this, there is no doubt that it can be obtained at speed of opening lists themselves. We are on the right way. 😉
